iosxe_prefix_list.tf

```diff
@@ -0,0 +1,34 @@
+locals {
+  prefix_lists = flatten([
+    for device in local.devices : [
+      for prefix_list in try(local.device_config[device.name].prefix_lists, []) : {
+        key    = format("%s/%s", device.name, prefix_list.name)
+        device = device.name
+
+        prefixes = [for e in try(prefix_list.seqs, []) : {
+          name   = prefix_list.name
+          seq    = e.seq
+          action = try(e.action, local.defaults.iosxe.configuration.prefix_lists.seqs.action, null)
+          ip     = try(e.ip, local.defaults.iosxe.configuration.prefix_lists.seqs.ip, null)
+          ge     = try(e.greater_equal, local.defaults.iosxe.configuration.prefix_lists.seqs.greater_equal, null)
+          le     = try(e.less_equal, local.defaults.iosxe.configuration.prefix_lists.seqs.less_equal, null)
+        }]
+
+        prefix_list_description = [for e in try(prefix_list, []) : {
```

Comment:
We are already iterating over prefix-lists in line #4. Therefore we can just construct a single element list:

```hcl
        prefix_list_description = [{
          name        = prefix_list.name
          description = try(prefix_list.description, local.defaults.iosxe.configuration.prefix_lists.description, null)
        }]
```

We should check if this works for prefix-lists without a description and if not add a condition to the whole element.